ponycheck/generator.pony

```diff
@@ -116,6 +116,37 @@ primitive Generators
             .collect[S](S.create(size))
       end)
 
+  fun set_of[T: (Hashable #read & Equatable[T] #read)](
+    gen: Generator[T],
+    max: USize = 100)
+    : Generator[Set[T]]
+  =>
+    """
+    Create a generator for sets filled with values
+    of the given generator ``gen``.
+    The returned sets will have a size between ``min`` and ``max``
```

Comment:
The documentation should probably say something along the lines of "returned sets will have a size up to ``max``" since there is no `min`.